homeassistant/components/legrand_rflc/config_flow.py

```diff
@@ -0,0 +1,171 @@
+"""Config flow for Legrand RFLC integration."""
+
+from __future__ import annotations
+
+import asyncio
+import logging
+import socket
+from typing import Any, Final
+
+import lc7001.aio
+import voluptuous
+
+from homeassistant import config_entries, data_entry_flow
+from homeassistant.components.dhcp import IP_ADDRESS
+from homeassistant.const import (
+    CONF_AUTHENTICATION,
+    CONF_HOST,
+    CONF_MAC,
+    CONF_PASSWORD,
+    CONF_PORT,
+)
+from homeassistant.helpers.typing import DiscoveryInfoType
+
+from .const import DOMAIN
+
+_LOGGER: Final = logging.getLogger(__name__)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """ConfigFlow for Legrand RFLC integration."""
+
+    VERSION = 1
+
+    HOST: Final = lc7001.aio.Connector.HOST
+
+    ABORT_NO_DEVICES_FOUND: Final = "no_devices_found"
+    ABORT_REAUTH_SUCCESSFUL: Final = "reauth_successful"
+
+    ERROR_INVALID_HOST: Final = "invalid_host"
+    ERROR_INVALID_AUTH: Final = "invalid_auth"
+
+    # Automatic ssdp discovery does not help.
+
+    # Automatic zeroconf discovery does not help.
+    # Even though the device supports mDNS, it does not publish any services to discover.
+
+    # Automatic dhcp discovery can detect conversations (udp port bootpc or bootps)
+    # with network devices that present their hostname as 'Legrand LC7001'
+    # with a manifest.json entry of
+    #   "dhcp": [{"hostname": "legrand lc7001"}]
+    # This will happen each time a Legrand LC7001 controller (re)boots.
+    # Linux requires the hass process have an effective cap_net_raw capability (or be run as root)
+    # for this to work.
+    async def async_step_dhcp(
+        self, discovery_info: DiscoveryInfoType
+    ) -> data_entry_flow.FlowResult:
+        """Handle a flow initiated by dhcp discovery."""
+        # example discovery_info
+        # {'ip': '192.168.0.1', 'hostname': 'legrand lc7001', 'macaddress': '0026ec000000'}
+        try:
+            resolutions = await asyncio.get_event_loop().getaddrinfo(self.HOST, None)
+        except OSError as error:
+            _LOGGER.warning("OS getaddrinfo %s error %s", self.HOST, error)
+            return self.async_abort(reason=self.ABORT_NO_DEVICES_FOUND)
+        address = discovery_info[IP_ADDRESS]
+        if any(
+            resolution[4][0] == address
+            for resolution in resolutions
+            if resolution[0] == socket.AF_INET
+        ):
```

Comment:
I am relying on DHCP discovery to tell me when it has seen a DHCP client with hostname "legrand lc7001".
I am not questioning that discovery process.
I am just doing further confirmation, at the time of the discovery, that any hostname resolution of the default HOST (LCM1.local) resolves to the same IP address.
A properly behaving Legrand RFLC LC7001 hub should pass both these tests.
If that happens then I will advance to the user step; otherwise the user will have to initiate adding the integration.